### DISCUSSION TITLE
کسی کانفیگ dokodemo-door  بلده درست کنه

### DISCUSSION THREAD

**Discussion by GFW-knocker** [GFW-knocker/gfw_resist_tls_proxy#47] [Q&A]
میخوام یه کانفیگ v2ray داشته باشیم که با dokodemo-door
 بصورت transparent proxy عمل کنه 
میخوام درخواستی که از سمت v2ray میاد برای سرور یوتیوب قابل فهم باشه
یعنی پروتکل vmess و اینا نباشه crypt نکنه که اونطرف خط  نمیخوایم دوباره xray داشته باشیم
من میخوام اونطرف سرور یوتیوب باشه
که یه کانفیگ بسازیم بدون سرور واسطه فقط برا یوتیوب

اگه بلدید باید یه فایل json بسازید و مثلا یه سایت معمولی مثل google.com بدید و با اون کانفیگ باز بشه
اگر کار کرد بدید  فرگمنت سوار کنیم روش  و  یوتیوب رو آزاد کنیم برای همیشه با ایپی های تمیز گوگل

**Comment by shaerpour** [comment]
پروتکل dokodemo-door کارش اینه که درخواست رو با هرشکلی دریافت میکنه میفرسته برای outbound که پایین تر تعریف شده در نتیجه بازم سرور دومی که خارج باشه نیازه که درخواست فرستاده بشه سمتش

{
  "log": {
    "loglevel": "warning"
  },
  "inbounds": [{
    "port": 1010,
    "protocol": "dokodemo-door",
    "settings": {
      "followRedirect": true
    },
    "sniffing": {
      "enabled": true,
      "destOverride": ["http", "tls"]
    }
  }],
  "outbounds": [
    {
      "tag": "proxy",
      "protocol": "vmess",
      "settings": {
        "vnext": [
          {
            "address": "<UPSTREAM-IP>",
            "port": 1310,
            "users": [
              {
                "id": "<UPSTREAM-UUID>",
                "alterId": 0,
                "security": "none"
              }
            ]
          }
        ]
      },
      "streamSettings": {
        "network": "ws"
      },
      "mux": {
        "enabled": true
      }
    },
    {
      "protocol": "freedom",
      "tag": "freedom"
    }
  ],
  "dns": {
    "servers": [
      "8.8.8.8",
      "8.8.4.4",
      "localhost"
    ]
  },
  "routing": {
    "domainStrategy": "IPIfNonMatch",
    "settings": {
      "rules": [
        {
          "type": "field",
          "outboundTag": "freedom",
          "domain": [
            "regexp:.*\\.ir$"
          ]
        }
      ]
    }
  }
}

همچین کانفیگی نیاز هست که بتونی با پورت ۱۰۱۰ که داده میشه کل خروجی رو بفرستی سمت outbound. البته نیازه رو خود سرور باید تنظیم شه که کل ترافیک سمت همین پورت ۱۰۱۰ که dokodemo داره گوش میده فرستاده بشه
